docs/adr/0009_plugin_registry.md

```diff
@@ -0,0 +1,306 @@
+directly, it contains references to individual plugin components.
+Like a table of contents that points to where all the plugins actually live:
+
+```yaml
+# Registry Component: ghcr.io/ocm/registry//ocm.software/plugin-registry:v1.0.0
+name: ocm.software/plugin-registry
+version: v1.0.0
+provider: ocm.software
+
+references:
+  - name: ecrplugin
+    version: 0.27.0
+    component: ocm.software/plugins/ecrplugin
+    digest:
+      hashAlgorithm: SHA-256
+      normalisationAlgorithm: jsonNormalisation/v1
+      value: abc123...
+  - name: helminput
+    version: 0.5.2
+    component: ocm.software/plugins/helminput
+    digest:
+      hashAlgorithm: SHA-256
+      normalisationAlgorithm: jsonNormalisation/v1
+      value: def456...
+  - name: cvedb
+    version: 1.2.0
+    component: enterprise.corp/plugins/cvedb
+    digest:
+      hashAlgorithm: SHA-256
+      normalisationAlgorithm: jsonNormalisation/v1
+      value: ghi789...
+
+resources:
+  - name: registry-metadata
+    type: blob
+    version: v1.0.0
+    access:
+      type: localBlob
+      localReference: sha256:...
+    # Contains plugin descriptions, categories, etc.
+```
+
+### Individual Plugin Component Structure
+
+Each plugin is just a regular OCM component version containing the actual plugin binaries for different platforms.
+This part doesn't change from how plugins work now:
+
+```yaml
+# Plugin Component: ghcr.io/ocm/registry//ocm.software/plugins/ecrplugin:0.27.0
+name: ocm.software/plugins/ecrplugin
+version: 0.27.0
+provider: ocm.software
+
+resources:
+  - name: demo                    # Plugin binary name
+    type: ocmPlugin
+    version: 0.27.0
+    extraIdentity:
+      architecture: amd64
+      os: linux
+    access:
+      type: localBlob
+      localReference: sha256:ac3f34...
+  - name: demo
+    type: ocmPlugin
+    version: 0.27.0
+    extraIdentity:
+      architecture: amd64
+      os: darwin
+    access:
+      type: localBlob
+      localReference: sha256:f1764c...
+  # ... additional platforms
+
+  - name: plugin-metadata        # Plugin description
+    type: blob
+    version: 0.27.0
+    access:
+      type: localBlob
+      localReference: sha256:...
+    # Contains: description, usage, capabilities, etc.
+```
+
+## User Workflow
+
+These are general flows regardless of the registry implementation.
+
+CLI command could be added that updated the user's configuration with the repository:
+
+```bash
+ocm plugin registry add official ghcr.io/ocm/registry//ocm.software/plugin-registry:v1.0.0
+```
+
+Or, the user can configure it manually:
+
+```yaml
+type: generic.config.ocm.software/v1
+configurations:
+- type: plugin.registry.config.ocm.software
+  registries:
+  - name: official
+    url: ghcr.io/ocm/registry//ocm.software/plugin-registry:v1.0.0
+  - name: enterprise
+    url: internal.corp/registry//enterprise.plugins/registry:latest
+    credentials:
```

Comment:
this should be part of the credential graph and not be part of the registry config, so a plugin registry looks up via a consumer identity